plugins/com.google.cloud.tools.eclipse.jdt.launching/src/com/google/cloud/tools/eclipse/jdt/internal/launching/SocketListenMultiConnectorProcess.java

```diff
@@ -0,0 +1,404 @@
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Google Inc - add support for accepting multiple connections
+ *******************************************************************************/
+package com.google.cloud.tools.eclipse.jdt.internal.launching;
+
+import com.sun.jdi.VMDisconnectedException;
+import com.sun.jdi.VirtualMachine;
+import com.sun.jdi.connect.Connector;
+import com.sun.jdi.connect.IllegalConnectorArgumentsException;
+import com.sun.jdi.connect.ListeningConnector;
+import com.sun.jdi.connect.TransportTimeoutException;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.IJobChangeEvent;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.jobs.JobChangeAdapter;
+import org.eclipse.debug.core.DebugEvent;
+import org.eclipse.debug.core.DebugException;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.model.IDebugTarget;
+import org.eclipse.debug.core.model.IProcess;
+import org.eclipse.debug.core.model.IStreamsProxy;
+import org.eclipse.jdi.TimeoutException;
+import org.eclipse.jdt.debug.core.JDIDebugModel;
+import org.eclipse.jdt.internal.launching.LaunchingMessages;
+import org.eclipse.jdt.internal.launching.LaunchingPlugin;
+import org.eclipse.jdt.internal.launching.SocketListenConnector;
+import org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants;
+import org.eclipse.osgi.util.NLS;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Fork of
+ * {@link org.eclipse.jdt.internal.launching.SocketListenConnectorProcess}.
+ * Knows how to handle "acceptCount" count.
+ * 
+ * A process that represents a VM listening connector that is waiting for a VM
+ * to remotely connect. Allows the user to see the status of the connection and
+ * terminate it. If a successful connection occurs, the debug target is added to
+ * the launch and this process is removed. Supports receiving multiple
+ * connections.
+ * 
+ * @since 3.4
+ * @see SocketListenConnector
+ */
+public class SocketListenMultiConnectorProcess implements IProcess {
+    /**
+     * The number of incoming connections to accept (0 = unlimited). This number
+     * is decremented after every accepted connection. If 0, technically means
+     * we can only accept up to |Integer.MIN_VALUE| connections, but that should
+     * be plenty for a debug connector.
+     */
+	final private AtomicInteger fAcceptCount;
```

Comment:
There are some comparisons of `fAcceptCount` that are done outside of the `WaitForConnectionJob`.  Rather than mark `fAcceptCount` as volatile, I figured I may as well use an `AtomicInteger` ⚛.

I've rewritten it to use a counter of the accepted connections, as I wondered if it might be helpful to label the VMs with their respective order.